client/txn.go

```diff
@@ -273,7 +284,16 @@ func (txn *Txn) Run(b *Batch) error {
 // efficient than relying on the implicit commit performed when the transaction
 // function returns without error.
 func (txn *Txn) Commit(b *Batch) error {
-	args := &proto.EndTransactionRequest{Commit: true}
+	return txn.endInBatch(b, true)
+}
+
+// Rollback a transaction.
+func (txn *Txn) Rollback(b *Batch) error {
```

Comment:
sorry, i think my previous comment wasn't right. Really the problem here is that rolling back a transaction and passing a batch doesn't make any sense. This method should probably be:
```go
func (txn *Txn) Rollback() error {
  return txn.send(proto.Call{
    Args: &proto.EndTransactionRequest{Commit: false},
    Reply: &proto.EndTransactionResponse{},
  })
}
```

it might make sense to share this (by parameterizing `Commit`) with `exec`.